src/neptune/new/handler.py

```diff
@@ -156,12 +166,15 @@ def assign(self, value, wait: bool = False) -> None:
         """
         with self._container.lock():
             attr = self._container.get_attribute(self._path)
-            if attr:
-                if isinstance(value, Handler):
-                    value = ValueCopy(value)
-                attr.process_assignment(value, wait)
-            else:
-                self._container.define(self._path, value, wait)
+            if not attr:
+                neptune_value = cast_value(value)
+                attr = ValueToAttributeVisitor(self._container, parse_path(self._path)).visit(neptune_value)
+                self._container.set_attribute(self._path, attr)
+                value = neptune_value  # TODO: we should clean some rules regarding type casting in assignment
```

Comment:
Casting value when type is not known (attribute does not exist) and when we already have attribute are completely different things.
We should address this problem soon.